tools/setup_helpers/cmake.py

```diff
@@ -31,6 +31,57 @@ def _mkdir_p(d):
 USE_NINJA = (not check_negative_env_flag('USE_NINJA') and
              which('ninja') is not None)
 
+def convert_cmake_value_to_python_value(cmake_value, cmake_type):
```

Comment:
Any changes here?

Reply:
I added a line number for error indication:

```diff
@@ -1,13 +1,12 @@
-@staticmethod
 def convert_cmake_value_to_python_value(cmake_value, cmake_type):
     r"""Convert a CMake value in a string form to a Python value.
 
     Arguments:
-      cmake_value (string): The CMake value in a string form (e.g., "ON", "OFF", "1").
-      cmake_type (string): The CMake type of :attr:`cmake_value`.
+        cmake_value (string): The CMake value in a string form (e.g., "ON", "OFF", "1").
+        cmake_type (string): The CMake type of :attr:`cmake_value`.
 
     Returns:
-      A Python value corresponding to :attr:`cmake_value` with type :attr:`cmake_type`.
+        A Python value corresponding to :attr:`cmake_value` with type :attr:`cmake_type`.
     """
 
     cmake_type = cmake_type.upper()
@@ -23,18 +22,17 @@
     else:  # Directly return the cmake_value.
         return cmake_value
 
-@staticmethod
-def _get_cmake_cache_variables(cmake_cache_file):
+def get_cmake_cache_variables_from_file(cmake_cache_file):
     r"""Gets values in CMakeCache.txt into a dictionary.
 
     Arguments:
-      cmake_cache_file: A CMakeCache.txt file object.
+        cmake_cache_file: A CMakeCache.txt file object.
     Returns:
-      dict: A ``dict`` containing the value of cached CMake variables.
+        dict: A ``dict`` containing the value of cached CMake variables.
     """
 
     results = dict()
-    for line in cmake_cache_file:
+    for i, line in enumerate(cmake_cache_file, 1):
         line = line.strip()
         if not line or line.startswith(('#', '//')):
             # Blank or comment line, skip
@@ -43,11 +41,11 @@
         # Space can also be part of variable name and value
         matched = re.match(r'(\S.*):\s*([a-zA-Z_-][a-zA-Z0-9_-]*)\s*=\s*(.*)', line)
         if matched is None:  # Illegal line
-            raise ValueError('Unexpected line in {}: {}'.format(repr(cmake_cache_file), line))
+            raise ValueError('Unexpected line {} in {}: {}'.format(i, repr(cmake_cache_file), line))
         variable, type_, value = matched.groups()
         if type_.upper() in ('INTERNAL', 'STATIC'):
             # CMake internal variable, do not touch
             continue
-        results[variable] = CMake.convert_cmake_value_to_python_value(value, type_)
+        results[variable] = convert_cmake_value_to_python_value(value, type_)
 
     return results
```